### ISSUE TITLE
Unreadable strings in tracebacks (yellow text on yellow background)

### ISSUE THREAD

**Issue by EtiennePelletier** [jupyterlab/jupyterlab#17576]
## Description
Strings in tracebacks are formatted with yellow text, but there are cases where they will also have a yellow background, which makes them impossible to read.
This affects the light, dark and dark (high contrast) themes.

![Image](https://github.com/user-attachments/assets/1f49ff9c-3c78-4796-ada5-03e20bdb3f33)

There exists a CSS rule in the dark themes to set the text black (introduced in #13249, and later copied with the addition of the high contrast theme), but it currently does not have precedence for the string (with yellow foreground color) due to the use of `:where`. It does apply correctly to the function names and parenthesis, which would be white if the rule is manually disabled:
![Image](https://github.com/user-attachments/assets/37f25c65-cf12-4d16-a74c-74d2674d76f1))

## Reproduce
Any python code rendering strings in a traceback where they would have a yellow background, such as:
```python
def test(string):
    raise

test('hello')
```

## Expected behavior
If the text was black on yellow background, it would be very easy to read.

I see a few ways of addressing this problem, other suggestions are welcome:
1. Add `!important` to the CSS rule fixing the illegible text in the dark themes and copy this rule in the light theme CSS.
2. Remove the `:where` CSS function to increase its precedence and copy this rule to the light theme.
3. **Recommended**:  Remove the `:where` CSS function to increase its precedence and move this rule in the base CSS file rather than duplicating it in the three separate theme files
4. _More complex_: Prevent the creation of HTML elements with both `ansi-yellow-fg`  and `ansi-yellow-bg` classes.

## Context

- Operating System and version: Ubuntu 24.04
- Browser and version: Firefox 138.0
- JupyterLab version: 4.5.0a0 (also tested and reproduced with 3.6.8) 

---

I would be happy to open a PR with one of the simpler fixes.

# 🐍 🤹‍♂

**Comment by krassowski** [comment]
I cannot reproduce this issue neither in Firefox nor in Chrome.

Firefox:

![Image](https://github.com/user-attachments/assets/29b3fed9-04c4-4f9a-983f-0e0aea5a2838)

![Image](https://github.com/user-attachments/assets/77d3281a-dd58-457d-aae6-830e422934f0)

Chrome:

![Image](https://github.com/user-attachments/assets/335bab54-b9ec-4558-9726-70c9e9e8abbe)

**Comment by krassowski** [comment]
The difference seems to be at the DOM level:

![Image](https://github.com/user-attachments/assets/312e5dc9-5d91-40c4-a182-279f8664cf47)

Is it maybe because of a new version of a dependency, i.e.e ipykernel/ipython sending different color of tracebacks?

**Comment by krassowski** [comment]
I am able to reproduce with IPython 9.x which sends different `traceback` compared to IPython 8.x:

```
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mRuntimeError\u001b[39m                              Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[2]\u001b[39m\u001b[32m, line 4\u001b[39m\n\u001b[32m      1\u001b[39m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[34mtest\u001b[39m(string):\n\u001b[32m      2\u001b[39m     \u001b[38;5;28;01mraise\u001b[39;00m\n\u001b[32m----> \u001b[39m\u001b[32m4\u001b[39m \u001b[43mtest\u001b[49m\u001b[43m(\u001b[49m\u001b[33;43m'\u001b[39;49m\u001b[33;43mhello\u001b[39;49m\u001b[33;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[2]\u001b[39m\u001b[32m, line 2\u001b[39m, in \u001b[36mtest\u001b[39m\u001b[34m(string)\u001b[39m\n\u001b[32m      1\u001b[39m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[34mtest\u001b[39m(string):\n\u001b[32m----> \u001b[39m\u001b[32m2\u001b[39m     \u001b[38;5;28;01mraise\u001b[39;00m\n",
      "\u001b[31mRuntimeError\u001b[39m: No active exception to reraise"
```

vs

```
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mRuntimeError\u001b[0m                              Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[2], line 4\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21mtest\u001b[39m(string):\n\u001b[1;32m      2\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m\n\u001b[0;32m----> 4\u001b[0m \u001b[43mtest\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mhello\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "Cell \u001b[0;32mIn[2], line 2\u001b[0m, in \u001b[0;36mtest\u001b[0;34m(string)\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21mtest\u001b[39m(string):\n\u001b[0;32m----> 2\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m\n",
      "\u001b[0;31mRuntimeError\u001b[0m: No active exception to reraise"
```

In IPython 9.x it is

```
[33;43mhello
```

which means:
- 33: set yellow foreground
- 43: set yellow background

whereas in IPython 8.x it was

```
[38;5;124;43mhello
```

which means:
- 38: set foreground to 124 (red-ish)
- 43: set yellow background

This looks like a bug in IPython, because no fragment of traceback should have both yellow background and foreground.

**Comment by EtiennePelletier** [comment]
Ah yes, I confirm this is with IPython 9, thank you for identifying the underlying IPython issue and opening an issue there.
